experimental/plugins/plugintypes/auditlog.go

```diff
@@ -61,6 +63,9 @@ type AuditLogTransactionRequest interface {
 	Headers() map[string][]string
 	Body() string
 	Files() []AuditLogTransactionRequestFiles
+	Args() *collections.ConcatKeyed // A string representation of all request agruments in the format 'k=v,'
+	Length() int32
+	UID() string
```

Comment:
What is the request UID in the generic context of a request? How does it differ from the transaction ID that we already have?
I see that you are using the ID in the Metadata of WebResourcesActivity, and then using the new UID() in the HttpRequest, so there might be a different meaning that I'm missing. Would be great to have a comment line here to explain that.

Reply:
Good catch!
Length was intended to capture the size of the request in bytes.  It should be logging transaction.variables.fullRequestLength here.

I struggled with deciding if transaction.id was better logged in WebResourcesActivity.Metadata.uid (The logging system-assigned unique identifier of an event instance) or WebResourcesActivity.HTTPRequest.uid (The unique identifier of the http request).  Now that you've got me looking at this again.  I think this actually does belong in the http request and not the metadata.

It looks like I had started to build this out but never completed it. I'll tidy this up and add some tests for them.

Reply:
Awesome, thanks!

Reply:
After looking through the code it appears `transaction.variables.FullRequestLength` is never populated with data.  If we're not currently capturing this information, would it be preferred that I simply remove the Length field from `AuditLogTransactionRequest` or simply comment it out with a note about future use?

Reply:
Hmmm... I would say use it (it will be 0 for now), but let's create an issue and followup PR with that data populated ;)